helper/users/dynamic/pool.go

```diff
@@ -60,11 +63,21 @@ type PoolConfig struct {
 	MaxUGID int
 }
 
+// disable will return true if either min or max is set to Disable (-1),
+// indicating the client should not enable the dynamic workload users
+// functionality
+func (p *PoolConfig) disable() bool {
+	return p.MinUGID == doNotEnable || p.MaxUGID == doNotEnable
+}
+
 // New creates a Pool with the given PoolConfig options.
 func New(opts *PoolConfig) Pool {
 	if opts == nil {
 		panic("bug: users pool cannot be nil")
```

Comment:
Why do you panic instead of returning an error?

Reply:
Each of these conditions shouldn't be possible in a normal nomad agent - the validation occurs well before here in the config parsing. The panics here just serve to remind test authors they're trying to set an invalid configuration.